client/views/admin/upgrade/UpgradePage/UpgradePage.tsx

```diff
@@ -0,0 +1,98 @@
+import { Throbber, Box } from '@rocket.chat/fuselage';
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
+
+import type { UpgradeTabVariants } from '../../../../../lib/getUpgradeTabType';
+import Page from '../../../../components/Page';
+import { useRouteParameter, useQueryStringParameter } from '../../../../contexts/RouterContext';
+import { useAbsoluteUrl } from '../../../../contexts/ServerContext';
+import UpgradePageError from '../UpgradePageError';
+
+const iframeStyle = { width: '100%', height: '100%' };
+
+const getUrl = (type: UpgradeTabVariants, date: string | undefined): string => {
+	switch (type) {
+		case 'goFullyFeatured':
+			return 'https://go.rocket.chat/i/upgrade-ce-1-unregistered';
+		case 'goFullyFeaturedRegistered':
+			return 'https://go.rocket.chat/i/upgrade-ce-1-registered';
+		case 'trialGold':
+			return `https://go.rocket.chat/i/upgrade-gold-trial${date ? `?date=${date}` : ''}`;
+		case 'trialEnterprise':
+			return `https://go.rocket.chat/i/upgrade-ee-trial${date ? `?date=${date}` : ''}`;
+		case 'upgradeYourPlan':
+			return 'https://go.rocket.chat/i/upgrade-ce-2';
+	}
+};
+
+type NavigationMessage = { goTo: string };
+
+const messageIsNavigation = (message: unknown): message is NavigationMessage => {
+	if (typeof message === 'object' && message !== null) {
+		return 'goTo' in message;
+	}
+
+	return false;
+};
+
+const getWindowMessagePath = (e: MessageEvent<string>): string | undefined => {
+	let parsedMessage = {};
+
+	try {
+		parsedMessage = JSON.parse(e.data);
+	} catch (error) {
+		return;
+	}
+
+	if (messageIsNavigation(parsedMessage)) {
+		return parsedMessage.goTo;
+	}
```

Comment:
```suggestion
	try {
		const parsedMessage = JSON.parse(e.data);
		if (messageIsNavigation(parsedMessage)) {
			return parsedMessage.goTo;
		}
	} catch (error) {
		return;
	}
```

Reply:
if we use this approach the catch block will be unnecessary and we're not going to be able to validate the `parsedMessage` 😞